src/storage/useBucket.ts

```diff
@@ -0,0 +1,78 @@
+'use client'
+
+import { useQuery, useQueryClient, type UseQueryOptions } from '@tanstack/react-query'
+import { Models } from 'appwrite'
+import { useEffect, useMemo } from 'react'
+import { useAppwrite } from 'react-appwrite'
+import type { StorageFileOperation } from 'react-appwrite/storage/types'
+
+/**
+ * Fetches a collection from a database.
+ * @param bucketId The bucket the files belong to.
+ * @param queries Queries to filter the bucket by.
+ * @param search Search term to filter the bucket by.
+ * @param options Options to pass to `react-query`.
+* @link [Appwrite Documentation](https://appwrite.io/docs/client/storage?sdk=web-default#storageListFiles)
+ */
+export function useBucket(
+    bucketId: string,
+    queries: string[] = [],
+    search?: string,
+    options?: UseQueryOptions<(Models.File)[], unknown, (Models.File)[]>
+) {
+  const { storage } = useAppwrite()
+  const queryClient = useQueryClient()
+  const queryKey = useMemo(() => ['appwrite', 'storage', bucketId, { queries }], [bucketId, queries])
+  const queryResult = useQuery({
+    queryKey,
+    queryFn: async () => {
+      const response = await storage.listFiles(bucketId, queries, search)
+
+      return response.files
+    },
+
+    onSuccess: files => {
+      for (const file of files) {
+        queryClient.setQueryData(['appwrite', 'storage', bucketId, 'files', file.$id], file)
+      }
+    },
+
+    ...options,
+  })
+
+  useEffect(() => {
+    const unsubscribe = storage.client.subscribe(`buckets.${bucketId}.files`, response => {
+      const [, operation] = response.events[0].match(/\.(\w+)$/) as RegExpMatchArray
+      const file = response.payload as Models.File
+
+      switch (operation as StorageFileOperation) {
+        case 'create':
+        case 'update':
+          queryClient.setQueryData(['appwrite', 'buckets', bucketId, 'file', file.$id], file)

```

Comment:
Can you change the query key here for the realtime updating so that it matches the query key `useFile` has? It should be updating `['appwrite', 'storage', bucketId, fileId]`

But honestly, the query keys throughout the project are kind of wild right now, and it's my fault. We will need a solution that is typesafe and consistent throughout all the services and their hooks.